Update ContainerDefinition.Secret valueFrom to pulumi.Input

gmaliar opened the pull request:
Hey team, 
I recently stumbled upon a use case of building a stack that also creates the secrets in secrets manager and `valueFrom` would fail for the `pulumi.Output<string>` type

An example use case to reproduce the error
```ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";

const testSecret = new aws.secretsmanager.Secret("test-secret", {
    namePrefix: "test-secret",
});

new aws.secretsmanager.SecretVersion("test-secret-version", {
    secretId: testSecret.id,
    secretString: "test",
});

const cluster = new awsx.ecs.Cluster("ecs-cluster");
const service = new awsx.ecs.FargateService("farget-service", {
    platformVersion: '1.3.0',
    cluster: cluster,
    desiredCount: 1,
    taskDefinitionArgs: {
        executionRole: executionRole,
        containers: {
            echoServer: {
                image: "busybox:latest",
                memory: 512,
                secrets: [
                    { name: "TEST", valueFrom: testSecret.arn }
                ]
            },
        },
    },
});
```

gmaliar commented:
@stack72 @jen20 can you point me to the contribution guide, I'm not sure if I'm missing anything to get this merged 😀

leezen commented:
@gmaliar Just realized I think we missed assigning a reviewer.
@gmaliar @stack72 Is there any reason why we wouldn't want to convert everything in the definition interface to be `Input<T>`?

gmaliar commented:
@leezen I guess what you are saying makes sense, would you like me to do that in this PR or an additional PR?

leezen commented:
@gmaliar If you don't mind turning this PR into that, happy to keep the review in here.

gmaliar commented:
Will do. 💪

CoderSpinoza commented:
@gmaliar Is there a reason this isn't merged?!

gmaliar commented:
@CoderSpinoza I forgot about this PR. Thanks for notifying me about it.

@leezen @stack72 I've rebased on top of the current main branch and refactored all interfaces to `pulumi.Input<T>`.
I wasn't sure what to do about `KernelCapability`, `LogDriver`, `Protocol` and `UlimitName`, my Typescript knowledge is still pretty limited, let me know how to tackle it.

leezen commented:
@gmaliar Thanks! I think those should be fine since either they're enums which are unlikely to come as outputs from other resources or are underlying types which you've converted to to also accepting `Input` which makes sense.

gmaliar commented:
@leezen that's what I thought but I wasn't sure.

leezen commented:
/run-acceptance-tests

CoderSpinoza commented:
@gmaliar wow, glad to see this is progressing. :) Hope it gets merged soon :)

gmaliar commented:
Waiting on the reviewer 🙏🙏🙏

leezen commented:
Sorry -- fairly sure this should be good, but I do want to have the tests run. I'm not sure why the last run got stuck.

leezen commented:
/run-acceptance-tests

leezen commented:
Ah, interesting, the previous run did succeed but somehow didn't post back here correctly: https://github.com/pulumi/pulumi-aws/runs/4956469406?check_suite_focus=true

leezen commented:
And here's the current run: https://github.com/pulumi/pulumi-aws/actions/runs/1780638789

gmaliar commented:
Thanks @leezen! It's a pleasure working with you.
What are the next steps now?

leezen commented:
@gmaliar Sorry about the delay. I need to work w/ the team to remove the status checks (at least the way they are). If I can't get that figured out soon-ish, I'll just override via admin and squash and merge.

stack72 commented:
Hi @gmaliar 

I am merging this now! Thanks for all the work here - sorry for the waiting you had to occur. Thanks to @leezen for getting this tested :)
